### PR TITLE
PIN-3536 [BE] Endpoint implementation for retrieving of the notices on backend for frontend

### PR DESCRIPTION
**Pull request by nttdata-rtorsoli**

### CODE REVIEW

File: src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/PrivacyNoticesApiServiceImpl.scala [hunk 1]
```diff
@@ -104,4 +109,28 @@ final case class PrivacyNoticesApiServiceImpl(
...
+
+  override def getPrivacyNoticeFromBucket(consentType: String)(implicit
+    contexts: Seq[(String, String)],
+    toEntityMarshallerProblem: ToEntityMarshaller[Problem],
+    toEntityMarshallerFile: ToEntityMarshaller[File]
+  ): Route = {
+    logger.info(s"Retrieving privacy notices for consentType $consentType")
+
+    val result: Future[HttpEntity.Strict] = for {
+      ctype  <- ConsentType.fromValue(consentType).toFuture
```
Comment (TonioGela) [inline src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/PrivacyNoticesApiServiceImpl.scala]: `ConsentType` is a really misleading name 😂

File: src/main/resources/interface-specification.yml [hunk 2]
```diff
@@ -8820,6 +8820,79 @@ paths:
...
+        - $ref: '#/components/parameters/IpAddress'
+        - name: consentType
+          in: path
+          required: true
+          schema:
+            $ref: '#/components/schemas/ConsentType'
+    get:
+        tags:
+          - privacyNotices
+        operationId: getPrivacyNoticeFromBucket
```
Comment (galales) [inline src/main/resources/interface-specification.yml]:
```suggestion
        operationId: getPrivacyNoticeContent
```

File: src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/PrivacyNoticesApiServiceImpl.scala [hunk 3]
```diff
@@ -104,4 +109,28 @@ final case class PrivacyNoticesApiServiceImpl(
...
+      ppId   <- consentTypeMap.get(ctype).toFuture(PrivacyNoticeNotFoundInConfiguration(consentType))
+      ppUuid <- ppId.toFutureUUID
```
Comment (TonioGela) [inline src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/PrivacyNoticesApiServiceImpl.scala]: I'll rather prefer long names, `ppId` is too hermetic, it's impossible to determine what is it

File: src/main/resources/interface-specification.yml [hunk 4]
```diff
@@ -8820,6 +8820,79 @@ paths:
...
+        description: Retrieve a specified version of privacy notice from AWS bucket
```
Comment (galales) [inline src/main/resources/interface-specification.yml]: No need to expose that kind of info in the openapi spec
```suggestion
        description: Retrieve the content of the privacy notice version
```

**Review by TonioGela — APPROVED** [review]

File: src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/PrivacyNoticesApiServiceImpl.scala (hunk 3 above)
Comment (galales) [inline src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/PrivacyNoticesApiServiceImpl.scala]: please use extended naming to improve readability
```suggestion
      consentType  <- ConsentType.fromValue(consentType).toFuture
      privacyNoticeId   <- consentTypeMap.get(consentType).toFuture(PrivacyNoticeNotFoundInConfiguration(consentType))
      privacyNoticeUuid <- privacyNoticeId.toFutureUUID
```

**Review by galales — APPROVED** [review]

**Review by beetlecrunch — APPROVED** [review]

**Review by galales — APPROVED** [review]

**Review by galales — APPROVED** [review]

**Pull request merged**